App crashes when instantiating HelpStack

We tried integrating HelpStack by following steps given on GitHub, but we kept getting the following errors:

04-03 13:54:22.054 4638-4638/com.playerline.android E/AndroidRuntime﹕ FATAL EXCEPTION: main
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.playerline.android/com.tenmiles.helpstack.activities.HomeActivity}: java.lang.NullPointerException
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:1967)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:1992)
at android.app.ActivityThread.access$600(ActivityThread.java:127)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1158)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:137)
at android.app.ActivityThread.main(ActivityThread.java:4448)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:511)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551)
at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
at com.tenmiles.helpstack.activities.HSActivityParent.onCreate(HSActivityParent.java:48)
at com.tenmiles.helpstack.activities.HomeActivity.onCreate(HomeActivity.java:46)
at android.app.Activity.performCreate(Activity.java:4465)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1049)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:1931)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:1992)
at android.app.ActivityThread.access$600(ActivityThread.java:127)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1158)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:137)
at android.app.ActivityThread.main(ActivityThread.java:4448)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:511)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551)
at dalvik.system.NativeStart.main(Native Method)

I'm guessing it has something to do with the App Theme, which in our case is Theme.AppCompat.NoActionBar. Should I be looking at some other area?

--- a/Themes/HSDarkTheme/values/hs_custom_theme.xml
+++ b/Themes/HSDarkTheme/values/hs_custom_theme.xml
@@ -67,16 +67,7 @@
     <style name="hs_messageEditTextStyle" parent="hs_editText_defaultStyle"></style> 
 	    
     <!-- DRAWABLES
-    Defining the drawable icons used in the project, override the drawable to give your own custom icons
-    
-    Attachment icon used in New Issue screen, and chat screen to show attachments -->
-    <drawable name="hs_attachment_icon">@drawable/hs_attachment_img</drawable>
-    
-   <!--  Search icon used in search menu -->
-    <drawable name="hs_search_icon">@drawable/hs_search_img_light</drawable>
-    
-    <!-- Disclosure icon used in main list view -->
-    <drawable name="hs_disclosure_next">@drawable/hs_disclosure_img</drawable>
+    Defining the drawable icons used in the project, override the drawable to give your own custom icons -->
     
     <!-- Add attachment icon used in issue detail screen - add reply -->
     <drawable name="hs_add_attachment">@drawable/hs_add_attachment_img</drawable>

--- a/Themes/HSFacebookTheme/values/hs_custom_theme.xml
+++ b/Themes/HSFacebookTheme/values/hs_custom_theme.xml
@@ -68,16 +68,7 @@
     <style name="hs_messageEditTextStyle" parent="hs_editText_defaultStyle"></style> 
 	    
     <!-- DRAWABLES
-    Defining the drawable icons used in the project, override the drawable to give your own custom icons
-    
-    Attachment icon used in New Issue screen, and chat screen to show attachments -->
-    <drawable name="hs_attachment_icon">@drawable/hs_attachment_img</drawable>
-    
-    <!--  Search icon used in search menu -->
-    <drawable name="hs_search_icon">@drawable/hs_search_img_light</drawable>
-    
-    <!-- Disclosure icon used in main list view -->
-    <drawable name="hs_disclosure_next">@drawable/hs_disclosure_img</drawable>
+    Defining the drawable icons used in the project, override the drawable to give your own custom icons -->
     
     <!-- Add attachment icon used in issue detail screen - add reply -->
     <drawable name="hs_add_attachment">@drawable/hs_add_attachment_img</drawable>

--- a/Themes/HSHalloweenTheme/values/hs_custom_theme.xml
+++ b/Themes/HSHalloweenTheme/values/hs_custom_theme.xml
@@ -67,16 +67,7 @@
     <style name="hs_messageEditTextStyle" parent="hs_editText_defaultStyle"></style> 
 	    
     <!-- DRAWABLES
-    Defining the drawable icons used in the project, override the drawable to give your own custom icons
-    
-    Attachment icon used in New Issue screen, and chat screen to show attachments -->
-    <drawable name="hs_attachment_icon">@drawable/hs_attachment_img</drawable>
-    
-   <!--  Search icon used in search menu -->
-    <drawable name="hs_search_icon">@drawable/hs_search_img_light</drawable>
-    
-    <!-- Disclosure icon used in main list view -->
-    <drawable name="hs_disclosure_next">@drawable/hs_disclosure_img</drawable>
+    Defining the drawable icons used in the project, override the drawable to give your own custom icons -->
     
     <!-- Add attachment icon used in issue detail screen - add reply -->
     <drawable name="hs_add_attachment">@drawable/hs_add_attachment_img</drawable>

--- a/Themes/HSLightTheme/values/hs_custom_theme.xml
+++ b/Themes/HSLightTheme/values/hs_custom_theme.xml
@@ -68,16 +68,7 @@
     <style name="hs_messageEditTextStyle" parent="hs_editText_defaultStyle"></style> 
 	    
     <!-- DRAWABLES
-    Defining the drawable icons used in the project, override the drawable to give your own custom icons
-    
-    Attachment icon used in New Issue screen, and chat screen to show attachments -->
-    <drawable name="hs_attachment_icon">@drawable/hs_attachment_img</drawable>
-    
-    <!--  Search icon used in search menu -->
-    <drawable name="hs_search_icon">@drawable/hs_search_img</drawable>
-    
-    <!-- Disclosure icon used in main list view -->
-    <drawable name="hs_disclosure_next">@drawable/hs_disclosure_img</drawable>
+    Defining the drawable icons used in the project, override the drawable to give your own custom icons -->
     
     <!-- Add attachment icon used in issue detail screen - add reply -->
     <drawable name="hs_add_attachment">@drawable/hs_add_attachment_img</drawable>

--- a/Themes/HSPathTheme/values/hs_custom_theme.xml
+++ b/Themes/HSPathTheme/values/hs_custom_theme.xml
@@ -68,16 +68,7 @@
     <style name="hs_messageEditTextStyle" parent="hs_editText_defaultStyle"></style> 
 	    
     <!-- DRAWABLES
-    Defining the drawable icons used in the project, override the drawable to give your own custom icons
-    
-    Attachment icon used in New Issue screen, and chat screen to show attachments -->
-    <drawable name="hs_attachment_icon">@drawable/hs_attachment_img</drawable>
-    
-   <!--  Search icon used in search menu -->
-    <drawable name="hs_search_icon">@drawable/hs_search_img_light</drawable>
-    
-    <!-- Disclosure icon used in main list view -->
-    <drawable name="hs_disclosure_next">@drawable/hs_disclosure_img</drawable>
+    Defining the drawable icons used in the project, override the drawable to give your own custom icons -->
     
     <!-- Add attachment icon used in issue detail screen - add reply -->
     <drawable name="hs_add_attachment">@drawable/hs_add_attachment_img</drawable>

--- a/Themes/HSPinterestTheme/values/hs_custom_theme.xml
+++ b/Themes/HSPinterestTheme/values/hs_custom_theme.xml
@@ -68,16 +68,7 @@
     <style name="hs_messageEditTextStyle" parent="hs_editText_defaultStyle"></style> 
 	    
     <!-- DRAWABLES
-    Defining the drawable icons used in the project, override the drawable to give your own custom icons
-    
-    Attachment icon used in New Issue screen, and chat screen to show attachments -->
-    <drawable name="hs_attachment_icon">@drawable/hs_attachment_img</drawable>
-    
-    <!--  Search icon used in search menu -->
-    <drawable name="hs_search_icon">@drawable/hs_search_img_light</drawable>
-    
-    <!-- Disclosure icon used in main list view -->
-    <drawable name="hs_disclosure_next">@drawable/hs_disclosure_img</drawable>
+    Defining the drawable icons used in the project, override the drawable to give your own custom icons -->
     
     <!-- Add attachment icon used in issue detail screen - add reply -->
     <drawable name="hs_add_attachment">@drawable/hs_add_attachment_img</drawable>

--- a/helpstack/res/layout/hs_activity_base.xml
+++ b/helpstack/res/layout/hs_activity_base.xml
@@ -1,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:orientation="vertical"
+    android:id="@+id/lyt_base"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+
+    <android.support.v7.widget.Toolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:visibility="gone">
+    </android.support.v7.widget.Toolbar>
+
+</LinearLayout>

--- a/helpstack/src/com/tenmiles/helpstack/activities/ArticleActivity.java
+++ b/helpstack/src/com/tenmiles/helpstack/activities/ArticleActivity.java
@@ -39,13 +39,12 @@ public class ArticleActivity extends HSActivityParent {
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
-		setContentView(R.layout.hs_activity_article);
+		setContentView(R.layout.hs_activity_article, savedInstanceState, R.string.hs_article);
 
 		if (savedInstanceState == null) {
 			HSKBItem kbItem = (HSKBItem)getIntent().getSerializableExtra("item");
 			ArticleFragment sectionFragment = HSFragmentManager.getArticleFragment(this, kbItem);
 			HSFragmentManager.putFragmentInActivity(this, R.id.container, sectionFragment, "Article");
-			getHelpStackActionBar().setTitle(R.string.hs_article);
 		}
 	}
 
@@ -64,10 +63,5 @@ public class ArticleActivity extends HSActivityParent {
 		}
 		return super.onOptionsItemSelected(item);
 	}
-	
-	@Override
-    public void configureActionBar(ActionBar actionBar) {
-    	super.configureActionBar(actionBar);
-    }
 
 }

--- a/helpstack/src/com/tenmiles/helpstack/activities/EditAttachmentActivity.java
+++ b/helpstack/src/com/tenmiles/helpstack/activities/EditAttachmentActivity.java
@@ -49,7 +49,7 @@ import com.tenmiles.helpstack.theme.widget.DrawingView;
 import java.io.FileNotFoundException;
 import java.util.UUID;
 
-public class EditAttachmentActivity extends AppCompatActivity {
+public class EditAttachmentActivity extends HSActivityParent {
 
     private final int REQUEST_CODE_PHOTO_PICKER = 100;
 
@@ -63,10 +63,7 @@ public class EditAttachmentActivity extends AppCompatActivity {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.hs_activity_edit_attachment);
-
-        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
-        getSupportActionBar().setTitle(R.string.hs_attachment_edit);
+        setContentView(R.layout.hs_activity_edit_attachment, savedInstanceState, R.string.hs_attachment_edit);
 
         drawView = (DrawingView)findViewById(R.id.drawing);
         drawView.setObserver(new DrawingView.ObserverInterface() {

--- a/helpstack/src/com/tenmiles/helpstack/activities/HSActivityParent.java
+++ b/helpstack/src/com/tenmiles/helpstack/activities/HSActivityParent.java
@@ -22,45 +22,70 @@
 
 package com.tenmiles.helpstack.activities;
 
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
 import android.os.Bundle;
 import android.support.v7.app.ActionBar;
 import android.support.v7.app.AppCompatActivity;
-import android.view.Window;
+import android.support.v7.widget.Toolbar;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.LinearLayout;
+
+import com.tenmiles.helpstack.R;
+import com.tenmiles.helpstack.gears.HSHappyfoxGear;
 
 /**
  * This is base class of all Activity used in HelpStack
- * 
- * @author Nalin Chhajer
  *
+ * @author Nalin Chhajer
  */
 public class HSActivityParent extends AppCompatActivity {
 
     private static final String ACTION_BAR_TITLE = "Actionbar_title";
-	
-	@Override
-	protected void onCreate(Bundle savedInstanceState) {
-		super.onCreate(savedInstanceState);
+    private Toolbar mToolbar;
 
-		// Handling actionbar title when activity changes so activity does not have to handle it.
-		if (savedInstanceState != null) {
-            getHelpStackActionBar().setTitle(savedInstanceState.getString(ACTION_BAR_TITLE));
-		}
-		getSupportActionBar().setHomeButtonEnabled(true);
-		getSupportActionBar().setDisplayHomeAsUpEnabled(true);
-		configureActionBar(getSupportActionBar());
-	}
+    // Handling actionbar title when activity changes so activity doesn't have to handle it.
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        if(getSupportActionBar() != null)
+        outState.putString(ACTION_BAR_TITLE, getSupportActionBar().getTitle().toString());
+    }
 
-	public void configureActionBar(ActionBar actionBar) {
-	}
-	
-	// Handling actionbar title when activity changes so activity doesnot have to handle it.
-	@Override
-	protected void onSaveInstanceState(Bundle outState) {
-		super.onSaveInstanceState(outState);
-		outState.putString(ACTION_BAR_TITLE, getHelpStackActionBar().getTitle().toString());
-	}
-	
-	public ActionBar getHelpStackActionBar() {
-		return getSupportActionBar();
-	}
+    protected void setContentView(int layoutResId, Bundle savedInstanceState, int title) {
+        super.setContentView(R.layout.hs_activity_base);
+        mToolbar =  (Toolbar)findViewById(R.id.toolbar);
+
+        LayoutInflater layoutInflater = LayoutInflater.from(this);
+        LinearLayout lLayout = (LinearLayout) findViewById(R.id.lyt_base);
+        View view = layoutInflater.inflate(layoutResId, null);
+
+        if(getSupportActionBar() != null){
+            ActionBar actionbar = getSupportActionBar();
+            setTitle(savedInstanceState, title);
+            actionbar.setHomeButtonEnabled(true);
+            actionbar.setDisplayHomeAsUpEnabled(true);
+        } else{
+            mToolbar.setVisibility(View.VISIBLE);
+            setSupportActionBar(mToolbar);
+            setTitle(savedInstanceState, title);
+        }
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+
+        lLayout.addView(view);
+    }
+
+    private void setTitle(Bundle savedInstanceState, int title) {
+        ActionBar actionbar = getSupportActionBar();
+        if (savedInstanceState != null) {
+            actionbar.setTitle(savedInstanceState.getString(ACTION_BAR_TITLE));
+        } else if(title != 0){
+            actionbar.setTitle(getString(title));
+        } else{
+            actionbar.setTitle(getString(R.string.hs_help_title));
+        }
+    }
 }

--- a/helpstack/src/com/tenmiles/helpstack/activities/HomeActivity.java
+++ b/helpstack/src/com/tenmiles/helpstack/activities/HomeActivity.java
@@ -43,7 +43,7 @@ public class HomeActivity extends HSActivityParent {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-		setContentView(R.layout.hs_activity_home);
+		setContentView(R.layout.hs_activity_home, savedInstanceState, R.string.hs_help_title);
 
         if (savedInstanceState == null) { // Activity started first time
         	HomeFragment homeFrag = HSFragmentManager.getHomeFragment();
@@ -51,12 +51,6 @@ public class HomeActivity extends HSActivityParent {
         }
     }
     
-    @Override
-    public void configureActionBar(ActionBar actionBar) {
-    	super.configureActionBar(actionBar);
-    	actionBar.setTitle(getString(R.string.hs_help_title)); 
-    }
-
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         int id = item.getItemId();

--- a/helpstack/src/com/tenmiles/helpstack/activities/ImageAttachmentDisplayActivity.java
+++ b/helpstack/src/com/tenmiles/helpstack/activities/ImageAttachmentDisplayActivity.java
@@ -38,7 +38,7 @@ public class ImageAttachmentDisplayActivity extends HSActivityParent {
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
-		setContentView(R.layout.hs_activity_image_attachment_display);
+		setContentView(R.layout.hs_activity_image_attachment_display, savedInstanceState, 0);
 
 		if (savedInstanceState == null) {
 			String url = getIntent().getExtras().getString(EXTRAS_STRING_URL);
@@ -47,11 +47,6 @@ public class ImageAttachmentDisplayActivity extends HSActivityParent {
             ImageAttachmentDisplayFragment fragment = HSFragmentManager.getImageAttachmentDisplayFragment(this, url);
             getSupportFragmentManager().beginTransaction().add(R.id.container, fragment).commit();
 		}
-	}
-	
-	@Override
-	public void configureActionBar(ActionBar actionBar) {
-		super.configureActionBar(actionBar);
 	}
 
 	@Override

--- a/helpstack/src/com/tenmiles/helpstack/activities/IssueDetailActivity.java
+++ b/helpstack/src/com/tenmiles/helpstack/activities/IssueDetailActivity.java
@@ -40,7 +40,7 @@ public class IssueDetailActivity extends HSActivityParent {
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
-		setContentView(R.layout.hs_activity_issue_detail);
+		setContentView(R.layout.hs_activity_issue_detail, savedInstanceState, 0);
 
 		if (savedInstanceState == null) {
 			IssueDetailFragment mIssueDetailFragment = HSFragmentManager.getIssueDetailFragment();
@@ -48,15 +48,10 @@ public class IssueDetailActivity extends HSActivityParent {
             HSFragmentManager.putFragmentInActivity(this, R.id.container, mIssueDetailFragment, TAG);
 			HSTicket ticket = (HSTicket)getIntent().getExtras().getSerializable(EXTRAS_TICKET);
 			mIssueDetailFragment.setTicket(ticket);
-			getHelpStackActionBar().setTitle(ticket.getSubject());
+			getSupportActionBar().setTitle(ticket.getSubject());
 		}
 	}
 
-	@Override
-	public void configureActionBar(ActionBar actionBar) {
-		super.configureActionBar(actionBar);
-	}
-	
 	@Override
 	public boolean onCreateOptionsMenu(Menu menu) {
 		getMenuInflater().inflate(R.menu.hs_issue_detail, menu);

--- a/helpstack/src/com/tenmiles/helpstack/activities/NewIssueActivity.java
+++ b/helpstack/src/com/tenmiles/helpstack/activities/NewIssueActivity.java
@@ -44,7 +44,7 @@ public class NewIssueActivity extends HSActivityParent {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.hs_activity_new_issue);
+        setContentView(R.layout.hs_activity_new_issue, savedInstanceState, R.string.hs_new_issue_title);
 
         if (savedInstanceState == null) {
             Bundle bundle = getIntent().getExtras();
@@ -57,12 +57,6 @@ public class NewIssueActivity extends HSActivityParent {
             }
             HSFragmentManager.putFragmentInActivity(this, R.id.container, newIssueFragment, "Issue");
         }
-    }
-
-    @Override
-    public void configureActionBar(ActionBar actionBar) {
-        super.configureActionBar(actionBar);
-        actionBar.setTitle(getString(R.string.hs_new_issue_title));
     }
 
     @Override

--- a/helpstack/src/com/tenmiles/helpstack/activities/NewUserActivity.java
+++ b/helpstack/src/com/tenmiles/helpstack/activities/NewUserActivity.java
@@ -37,20 +37,14 @@ public class NewUserActivity extends HSActivityParent {
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
-		setContentView(R.layout.hs_activity_new_user);
-		
+		setContentView(R.layout.hs_activity_new_user, savedInstanceState, R.string.hs_new_issue_title);
+
 		if (savedInstanceState == null) {
             Bundle bundle = getIntent().getExtras();
             NewUserFragment newUserFragment = new NewUserFragment();
             newUserFragment.setArguments(bundle);
             HSFragmentManager.putFragmentInActivity(this, R.id.container, newUserFragment, TAG);
         }
-	}
-	
-	@Override
-	public void configureActionBar(ActionBar actionBar) {
-		super.configureActionBar(actionBar);
-		actionBar.setTitle(getString(R.string.hs_new_issue_title));
 	}
 
 	@Override

--- a/helpstack/src/com/tenmiles/helpstack/activities/SectionActivity.java
+++ b/helpstack/src/com/tenmiles/helpstack/activities/SectionActivity.java
@@ -46,13 +46,13 @@ public class SectionActivity extends HSActivityParent {
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
-		setContentView(R.layout.hs_activity_section);
+		setContentView(R.layout.hs_activity_section, savedInstanceState, 0);
 
 		if (savedInstanceState == null) {
 			HSKBItem kbSectionItem = (HSKBItem)getIntent().getSerializableExtra(EXTRAS_SECTION_ITEM);
 			SectionFragment sectionFragment = HSFragmentManager.getSectionFragment(this, kbSectionItem);
             HSFragmentManager.putFragmentInActivity(this, R.id.container, sectionFragment, TAG);
-			getHelpStackActionBar().setTitle(kbSectionItem.getSubject());
+			getSupportActionBar().setTitle(kbSectionItem.getSubject());
 		}
 	}
 
@@ -70,11 +70,6 @@ public class SectionActivity extends HSActivityParent {
 			return true;
 		}
 		return super.onOptionsItemSelected(item);
-	}
-	
-	@Override
-	public void configureActionBar(ActionBar actionBar) {
-		super.configureActionBar(actionBar);
 	}
 
 }

--- a/helpstack/src/com/tenmiles/helpstack/fragments/ArticleFragment.java
+++ b/helpstack/src/com/tenmiles/helpstack/fragments/ArticleFragment.java
@@ -33,51 +33,51 @@ import com.tenmiles.helpstack.R;
 import com.tenmiles.helpstack.model.HSKBItem;
 
 public class ArticleFragment extends HSFragmentParent {
-	
-	public static final String HTML_WRAPPER_WITH_TITLE = 
-			"<!DOCTYPE html><html><head>" +
-			"</head><body><h3 class='heading'>%s</h3>%s</body></html>";
 
-	public HSKBItem kbItem;
-	private WebView webview;
-	
-	public ArticleFragment() {
-	}
-	
-	@Override
+    public static final String HTML_WRAPPER_WITH_TITLE =
+            "<!DOCTYPE html><html><head>" +
+                    "</head><body><h3 class='heading'>%s</h3>%s</body></html>";
+
+    public HSKBItem kbItem;
+    private WebView webview;
+
+    public ArticleFragment() {
+    }
+
+    @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-		View rootView = inflater.inflate(R.layout.hs_fragment_article, container, false);
-		
-		webview = (WebView)rootView.findViewById(R.id.webview);
-		
-		if (savedInstanceState != null) {
-			kbItem = (HSKBItem) savedInstanceState.getSerializable("kbItem");
-		}
-		
-		webview.setWebChromeClient(new WebChromeClient() {
-			   public void onProgressChanged(WebView view, int progress) {
-			     // Activities and WebViews measure progress with different scales.
-			     // The progress meter will automatically disappear when we reach 100%
-				   progress = progress*100;
-				   getHelpStackActivity().setSupportProgressBarVisibility(progress<98);
-				   getHelpStackActivity().setSupportProgress(progress);
-			   }
-			 });
-		
-		initializeView();
-		return rootView;
-	}
-	
-	@Override
-	public void onSaveInstanceState(Bundle outState) {
-		super.onSaveInstanceState(outState);
-		outState.putSerializable("kbItem", kbItem);
-	}
-	
-	public void initializeView() {
-		String content = this.kbItem.getBody();
-		String contentText = String.format(HTML_WRAPPER_WITH_TITLE, this.kbItem.getSubject(), content);
-		webview.loadData(contentText, "text/html", null);
-	}
+        View rootView = inflater.inflate(R.layout.hs_fragment_article, container, false);
+
+        webview = (WebView) rootView.findViewById(R.id.webview);
+
+        if (savedInstanceState != null) {
+            kbItem = (HSKBItem) savedInstanceState.getSerializable("kbItem");
+        }
+
+        webview.setWebChromeClient(new WebChromeClient() {
+            public void onProgressChanged(WebView view, int progress) {
+                // Activities and WebViews measure progress with different scales.
+                // The progress meter will automatically disappear when we reach 100%
+                progress = progress * 100;
+                getHelpStackActivity().setSupportProgressBarVisibility(progress < 98);
+                getHelpStackActivity().setSupportProgress(progress);
+            }
+        });
+
+        initializeView();
+        return rootView;
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putSerializable("kbItem", kbItem);
+    }
+
+    public void initializeView() {
+        String content = this.kbItem.getBody();
+        String contentText = String.format(HTML_WRAPPER_WITH_TITLE, this.kbItem.getSubject(), content);
+        webview.loadData(contentText, "text/html", null);
+    }
 
 }

--- a/helpstack/src/com/tenmiles/helpstack/fragments/HSFragmentParent.java
+++ b/helpstack/src/com/tenmiles/helpstack/fragments/HSFragmentParent.java
@@ -22,6 +22,8 @@
 
 package com.tenmiles.helpstack.fragments;
 
+import android.app.Activity;
+import android.content.Context;
 import android.support.v4.app.Fragment;
 import android.support.v7.app.ActionBar;
 
@@ -36,13 +38,21 @@ import com.tenmiles.helpstack.activities.HSActivityParent;
  */
 public class HSFragmentParent extends Fragment {
 
-	ActionBar getActionBar() {
-		HSActivityParent act = (HSActivityParent) getActivity();
-		return act.getSupportActionBar();
-	}
-	
+	HSActivityParent mActivity;
+
 	public HSActivityParent getHelpStackActivity() {
-		HSActivityParent act = (HSActivityParent) getActivity();
-		return act;
+		return mActivity;
+	}
+
+	@Override
+	public void onAttach(Context context) {
+		super.onAttach(context);
+		mActivity = (HSActivityParent) context;
+	}
+
+	@Override
+	public void onDetach() {
+		super.onDetach();
+		mActivity = null;
 	}
 }

